### ISSUE TITLE
[BUG] The position makes the light and shadow of the block dark

### ISSUE THREAD

**Issue by Caishangqi** [PluginBugs/Issues-ItemsAdder#1092]
**Describe the bug**
When I make a block and use REAL_TRANSPARENT, the block will turn black when it is on the ground but not if it is placed on the barrier.

**To Reproduce**
1. Create a transparent real block with a custom model
2. place the block on the ground
3. place the black on the barrier
4. the block on the ground would turn dark and the barrier one won't

**Expected behavior**
The block should not turn dark when place on ground.

**Screenshots**
![I)S3_C4@_ _9TXZ5~DMFN`J](https://user-images.githubusercontent.com/39553613/128609160-166a9c1c-6716-4c28-b3c8-9d83d986aff9.png)

**Info (please complete the following information):**
 - Server version: Paper 1.17.1
 - Client version: 1.17.1 Optfine G9 pre 29

**Special Info**
 - ItemsAdder version: 2.4.7
 - LightAPI Fork version: 3.5.2
 - LoneLibs version: 1.0.14
 - ProtocolLib version: 4.7.0

**Additional context**
The block [yaml](https://paste.ubuntu.com/p/99N5GZRNNX/)
The block [model ](https://paste.ubuntu.com/p/c8xvRfsfjJ/)

**Comment by LoneDev6** [comment]
I suppose you just need to move the model (inside BlockBench or modelling software) of 1-2 pixels up. This will avoid the model from being touched completely by the bottom block.

**Comment by Caishangqi** [comment]
> I suppose you just need to move the model (inside BlockBench or modelling software) of 1-2 pixels up. This will avoid the model from being touched completely by the bottom block.

I had moved the model 1-2 pixels up, it still has the problem.
![O8YJ%V@B1ZZ@(_WCB(MAC88](https://user-images.githubusercontent.com/39553613/128685861-741157e2-fe0d-484a-bf4b-cf1831d3b964.png)

**Comment by LoneDev6** [comment]
I think it's due to how Minecraft manages occlusion with bottom blocks.
Basically if the model is too high it will be back on the top.
You can try adding this attribute:
`	"ambientocclusion": false,`

```json
{
   "credit":"Made with Blockbench",
   "ambientocclusion": false,
   "textures":{
      "0":"blocks:block/lamppost",
      "particle":"blocks:block/lamppost"
   },
```

**Comment by LoneDev6** [comment]
only this model has this problem?
Do the same thing happens if you create another model with REAL_TRASPARENT?

**Comment by Caishangqi** [comment]
Sure, Only that model I with REA:_TRANSPARENT have this problem.
![@JEFW(Q8L) _IAOPS32CBNV](https://user-images.githubusercontent.com/39553613/128703225-87aa5142-53d3-4cf0-badc-1af8e2648d57.png)

[This one](https://paste.ubuntu.com/p/jRshSqpwVk/) works fine

**Comment by LoneDev6** [comment]
Can you try to change the texture of the bugged model and use the texture of the working model?
Let me know what happens.

**Comment by Caishangqi** [comment]
> Can you try to change the texture of the bugged model and use the texture of the working model?
> Let me know what happens.
> 
> Anyway I think it's because they have different heights, I'm sure if you make the working model higher it will have the same issue as the first one. (try)

The only difference between them is that the range of the texture used is different. I tried to zoom the model to a certain range in "Edit Mode", the shadow will disappear. But I don't know why I can't apply this model to the exact position when modifying the "display adjustment".
![M}PLE$6M)IMYZZO0HOPDM1U](https://user-images.githubusercontent.com/39553613/128757179-5b4d63a1-ef8c-4ad3-ade5-d84728f5d00a.png)

In the Adjustment interface:
![LI}LIY5KJSRZL3_3W0L_3W5](https://user-images.githubusercontent.com/39553613/128757280-3feccfdf-4d98-4b87-a7be-2f11c5cc628c.png)

Model show in Game:
![F04RN@B$9%3)U}2C FU0Q1J](https://user-images.githubusercontent.com/39553613/128757440-83c6da62-429a-4be1-b1be-eb1a5e4d0f84.png)
In addition, no matter how I try, the hitbox of this box is always 1-2 pixels above the ground

**Comment by LoneDev6** [comment]
I think we're losing track of the initial issue.
I see for some reason you're editing the position using the armor_stand view, which is useless if your model is a block (REAL_TRANSPARENT).
You have to adjust it in the main view

**Comment by LoneDev6** [comment]
Feel free to open a ticket on Discord for faster communication https://support.devs.beer/

**Comment by Caishangqi** [comment]
> Feel free to open a ticket on Discord for faster communication https://support.devs.beer/

okay, thanks :)

**Comment by LoneDev6** [comment]
Did you manage to solve the issue? :D 
Feel free to contact me on Discord if you want to try some solutions directly with me

**Comment by Caishangqi** [comment]
> Did you manage to solve the issue? :D
> Feel free to contact me on Discord if you want to try some solutions directly with me
Yes, I split the 2x1blocks into 2 1x1 blocks which solve the problems :)

**Comment by LoneDev6** [comment]
I think you can use the new feature of the upcoming update

"shift_up" attribute.
https://github.com/PluginBugs/Issues-ItemsAdder/issues/1110#issuecomment-901262262
![immagine](https://user-images.githubusercontent.com/27242001/130069096-87ef6473-dd8e-41d9-b182-96d7357fa9fd.png)


It works for all block types (obv not with FIRE)

**Comment by Caishangqi** [comment]
> I think you can use the new feature of the upcoming update, so you can use only 1 block instead of 2 blocks.
> 
> "shift_up" attribute.
> [#1110 (comment)](https://github.com/PluginBugs/Issues-ItemsAdder/issues/1110#issuecomment-901262262)
> ![immagine](https://user-images.githubusercontent.com/27242001/130069096-87ef6473-dd8e-41d9-b182-96d7357fa9fd.png)
> 
> It works for all block types (obv not with FIRE).

Thanks, I would try it in the new version of Itemsadder, it seems like this new feature could make aquatic plants!

**Comment by LoneDev6** [comment]
Yep, surface (aquatic plants).
Check this plastic waste example:
https://www.spigotmc.org/resources/deco-worlddeco-add-autogenerating-decorations-on-your-world-surface.95207/
![immagine](https://user-images.githubusercontent.com/27242001/130071056-a5edabdb-9ea3-4ada-9628-d21e33dc1405.png)

**Comment by Caishangqi** [comment]
> Yep, surface (aquatic plants).
> Check this plastic waste example:
> https://www.spigotmc.org/resources/deco-worlddeco-add-autogenerating-decorations-on-your-world-surface.95207/
> ![immagine](https://user-images.githubusercontent.com/27242001/130071056-a5edabdb-9ea3-4ada-9628-d21e33dc1405.png)

Done, works totally fine :D
![BG1AZ$3)BKS8`) CP)V7KU1](https://user-images.githubusercontent.com/39553613/130079313-763aedcf-4c3d-4efb-bbd2-1924f3b9a52e.png)

**Comment by LoneDev6** [comment]
Very good!
If you want to make it solid you can use REAL_TRANSPARENT and set shift_up property.
You have to open blockbench and move the model down like that.
![immagine](https://user-images.githubusercontent.com/27242001/130081914-a198ae62-d3a5-4660-9e8f-701b5fc0a6e3.png)

If you want to keep it non-solid but move the hitbox at the center you can just keep using REAL_WIRE and do the same thing as stated before (move the model down in blockbench).

**Comment by Caishangqi** [comment]
I had set the model in the middle and it was worked, but I am quite confused about how to use the attribute " shift_up" and "placeable_on_water", do those attributes use like this?
https://paste.ubuntu.com/p/g3SgQx3Qj4/

**Comment by LoneDev6** [comment]
Basically shift_up will place the model but 1 block up.
`shift_up: true` it doesn't accept digits, so `shift_up: 1` is wrong.

`placeable_on_water` will allow you to place the block MANUALLY on water. You don't need to specify it if you want to be generated on water (using surface_populators).

The shift_up attribute is shown in the new item added in the [WorldDeco update](https://www.spigotmc.org/resources/deco-worlddeco-add-autogenerating-decorations-on-your-world-surface.95207/updates)
![immagine](https://user-images.githubusercontent.com/27242001/130088322-33a919c7-6967-421b-ac52-bcf797ee9cff.png)

**Comment by Caishangqi** [comment]
> Basically shift_up will place the model but 1 block up.
> It's useful for blocks that have their model tall (but shifted down)
> `shift_up: true` it doesn't accept digits, so `shift_up: 1` is wrong.
> 
> ![immagine](https://user-images.githubusercontent.com/27242001/130088630-588732fb-f285-484c-b500-8b2d7513ecf0.png)
> 
> Without shift_up the model would be placed like that:
> ![immagine](https://user-images.githubusercontent.com/27242001/130088737-902bfa00-ee4f-440d-97a0-fee2549121b2.png)
> ![immagine](https://user-images.githubusercontent.com/27242001/130088848-84e929cf-f576-4c34-b5df-539f4cd78f22.png)
> 
> `placeable_on_water` will allow you to place the block MANUALLY on water. You don't need to specify it if you want to be generated on water (using surface_populators).
> 
> The shift_up attribute is shown in the new item added in the [WorldDeco update](https://www.spigotmc.org/resources/deco-worlddeco-add-autogenerating-decorations-on-your-world-surface.95207/updates)

Thanks for providing an example. it works!
And have a nice day :3
